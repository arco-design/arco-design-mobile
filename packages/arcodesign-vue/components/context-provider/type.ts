export interface ContextInstallOptions {
    /**
     * 组件名称前缀
     * @en Component prefix
     * @default "Arco"
     */
    componentPrefix?: string;
}

export interface GlobalContextParams {
    /**
     * 组件类名前缀
     * @en Component classname prefix
     * @default "arco"
     */
    prefixCls?: string;
}
