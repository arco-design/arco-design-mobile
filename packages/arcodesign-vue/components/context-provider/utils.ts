import { InjectionKey, inject } from 'vue';
import { ContextInstallOptions, GlobalContextParams } from './type';

export const defaultInstallOptions: Required<ContextInstallOptions> = {
    componentPrefix: 'Arco',
};

export const defaultContext: Required<GlobalContextParams> = {
    prefixCls: 'arco',
};

export const configProviderInjectionKey: InjectionKey<GlobalContextParams> =
    Symbol('ArcoContextProvider');

export function getComponentPrefix(options?: ContextInstallOptions) {
    return options?.componentPrefix || defaultInstallOptions.componentPrefix;
}

export function getPrefixCls(componentName?: string) {
    const contextProvider = inject(configProviderInjectionKey, defaultContext);
    const prefix = contextProvider.prefixCls;
    if (componentName) {
        return `${prefix}-${componentName}`;
    }
    return prefix;
}
