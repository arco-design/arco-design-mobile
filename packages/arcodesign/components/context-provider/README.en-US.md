### Others

# ContextProvider 

Global data control component, used to replace global data.

======

> Props

|Property|Description|Type|DefaultValue|
|----------|-------------|------|------|
|prefixCls|Component classname prefix|string|"arco"|
|system|Manually control the current system, and the incoming value will be used directly after being passed in\. It is applicable when the initial value of the system needs to be specified in the ssr scenario\.|"" \| "pc" \| "android" \| "ios"|""|
|useDarkMode|Whether to use dark mode|boolean|false|
|isDarkMode|Whether it is in dark mode|boolean|false|
|theme|Theme variable\. The css variable will be replaced online after input\. The less variable needs to be set|Record\<string, string\>|-|
|locale|Internationalized language configuration|ILocale|-|
|useRtl|Whether to use rtl|boolean|false|

> ILocale

|Property|Description|Type|
|----------|-------------|------|
|locale|Language Type|string|
|LoadMore|-|\{ loadMoreText: string; loadingText: string; prepareText: string; noDataText: string; failLoadText: string; prepareScrollText: string; prepareClickText: string; \}|
|Picker|-|\{ okText: string; cancelText: string; \}|
|Tag|-|\{ addTag: string; \}|
|Dialog|-|\{ okText: string; cancelText: string; \}|
|SwipeLoad|-|\{ normalText: string; activeText: string; \}|
|PullRefresh|-|\{ loadingText: string; pullingText: string; finishText: string; loosingText: string; \}|
|DropdownMenu|-|\{ select: string; \}|
|Pagination|-|\{ previousPage: string; nextPage: string; \}|
|Image|-|\{ loadError: string; \}|
|ImagePicker|-|\{ loadError: string; \}|
|SearchBar|-|\{ placeholder: string; cancelBtn: string; \}|
|Stepper|-|\{ minusButtonName: string; addButtonName: string; \}|
|Keyboard|-|\{ confirm: string; \}|
|Form|-|\{ required: string; type: \{ email: string; url: string; string: string; number: string; array: string; object: string; boolean: string; \}; number: \{ min: string; max: string; equal: string; range: string; positive: string; negative: string; \}; string: \{ \.\.\.; \}; array: \{ \.\.\.; \}; object: \{ \.\.\.; \}; boolean: \{ \.\.\.; \}; \}|
