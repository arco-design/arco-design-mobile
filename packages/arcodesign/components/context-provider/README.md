### 其他

# 全局配置 ContextProvider

全局数据控制组件，用于替换全局数据。

======

> 属性/Props

|参数|描述|类型|默认值|
|----------|-------------|------|------|
|prefixCls|组件类名前缀|string|"arco"|
|system|手动控制当前所在系统，传入后将直接使用传入的值，ssr场景需指定系统初始值时适用|"" \| "pc" \| "android" \| "ios"|""|
|useDarkMode|是否使用暗黑模式|boolean|false|
|isDarkMode|是否处于暗黑模式，指定后以指定的值为准|boolean|false|
|theme|主题变量，传入后将在线替换css变量，需设置less变量 @use\-css\-vars: 1|Record\<string, string\>|-|
|locale|国际化语言包配置|ILocale|-|
|useRtl|是否使用Rtl模式|boolean|false|

> ILocale

|参数|描述|类型|
|----------|-------------|------|
|locale|语言类型|string|
|LoadMore|-|\{ loadMoreText: string; loadingText: string; prepareText: string; noDataText: string; failLoadText: string; prepareScrollText: string; prepareClickText: string; \}|
|Picker|-|\{ okText: string; cancelText: string; \}|
|Tag|-|\{ addTag: string; \}|
|Dialog|-|\{ okText: string; cancelText: string; \}|
|SwipeLoad|-|\{ normalText: string; activeText: string; \}|
|PullRefresh|-|\{ loadingText: string; pullingText: string; finishText: string; loosingText: string; \}|
|DropdownMenu|-|\{ select: string; \}|
|Pagination|-|\{ previousPage: string; nextPage: string; \}|
|Image|-|\{ loadError: string; \}|
|ImagePicker|-|\{ loadError: string; \}|
|SearchBar|-|\{ placeholder: string; cancelBtn: string; \}|
|Stepper|-|\{ minusButtonName: string; addButtonName: string; \}|
|Keyboard|-|\{ confirm: string; \}|
|Form|-|\{ required: string; type: \{ email: string; url: string; string: string; number: string; array: string; object: string; boolean: string; \}; number: \{ min: string; max: string; equal: string; range: string; positive: string; negative: string; \}; string: \{ \.\.\.; \}; array: \{ \.\.\.; \}; object: \{ \.\.\.; \}; boolean: \{ \.\.\.; \}; \}|
