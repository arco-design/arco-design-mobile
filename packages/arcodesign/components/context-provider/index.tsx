import React, { createContext, useState, useEffect, useCallback } from 'react';
import { addCssRules, removeCssStyleDom, defaultLocale, ILocale } from '@arco-design/mobile-utils';

export interface GlobalContextParams {
    /**
     * 组件类名前缀
     * @en Component classname prefix
     * @default "arco"
     */
    prefixCls?: string;
    /**
     * 手动控制当前所在系统，传入后将直接使用传入的值，ssr场景需指定系统初始值时适用
     * @en Manually control the current system, and the incoming value will be used directly after being passed in. It is applicable when the initial value of the system needs to be specified in the ssr scenario.
     * @default ""
     */
    system?: 'pc' | 'android' | 'ios' | '';
    /**
     * 是否使用暗黑模式
     * @en Whether to use dark mode
     * @default false
     */
    useDarkMode?: boolean;
    /**
     * 是否处于暗黑模式，指定后以指定的值为准
     * @en Whether it is in dark mode
     * @default false
     */
    isDarkMode?: boolean;
    /**
     * 主题变量，传入后将在线替换css变量，需设置less变量 @use-css-vars: 1
     * @en Theme variable. The css variable will be replaced online after input. The less variable needs to be set @use-css-vars: 1
     */
    theme?: Record<string, string>;
    /**
     * 国际化语言包配置
     * @en Internationalized language configuration
     */
    locale?: ILocale;
    /**
     * 是否使用Rtl模式
     * @en Whether to use rtl
     * @default false
     */
    useRtl?: boolean;
}

export const defaultContext: GlobalContextParams = {
    prefixCls: 'arco',
    system: '',
    useDarkMode: false,
    isDarkMode: false,
    locale: defaultLocale,
    useRtl: false,
};

export const GlobalContext = createContext<GlobalContextParams>(defaultContext);

export interface ContextProviderProps extends GlobalContextParams {
    children: React.ReactNode;
}

export type WithGlobalContext<T> = T & { context?: GlobalContextParams };

/**
 * 全局数据控制组件，用于替换全局数据。
 * @en Global data control component, used to replace global data.
 * @type 其他
 * @type_en Others
 * @name 全局配置
 * @name_en ContextProvider
 */
export default function ContextProvider(props: ContextProviderProps) {
    const {
        children,
        useDarkMode,
        isDarkMode,
        theme,
        locale = defaultLocale,
        ...restProps
    } = props;
    const [isDarkModeState, setIsDarkModeState] = useState<boolean>();
    const changeDarkMode = useCallback(
        (res: MediaQueryListEvent) => setIsDarkModeState(res.matches),
        [],
    );

    useEffect(() => {
        const matchMedia = window.matchMedia('(prefers-color-scheme: dark)');
        if (useDarkMode) {
            const dark = matchMedia.matches;
            setIsDarkModeState(dark);
            if (typeof matchMedia.addEventListener === 'function') {
                matchMedia.addEventListener('change', changeDarkMode);
            } else if (typeof matchMedia.addListener === 'function') {
                matchMedia.addListener(changeDarkMode);
            }
        }
        return () => {
            if (useDarkMode) {
                if (typeof matchMedia.removeEventListener === 'function') {
                    matchMedia.removeEventListener('change', changeDarkMode);
                } else if (typeof matchMedia.removeListener === 'function') {
                    matchMedia.removeListener(changeDarkMode);
                }
            }
        };
    }, [useDarkMode]);

    useEffect(() => {
        if (!theme) {
            return;
        }
        addCssRules('arcoCustomTheme', theme);
        return () => {
            removeCssStyleDom('arcoCustomTheme');
        };
    }, [theme]);

    return (
        <GlobalContext.Provider
            value={{
                ...defaultContext,
                ...restProps,
                isDarkMode: isDarkMode === void 0 ? isDarkModeState : isDarkMode,
                useDarkMode,
                locale: locale || defaultLocale,
            }}
        >
            {children}
        </GlobalContext.Provider>
    );
}

export const ContextLayout = GlobalContext.Consumer;

export function CompWithGlobalContext<P extends JSX.IntrinsicAttributes>(
    Component: React.FunctionComponent<P>,
) {
    return function (props: WithGlobalContext<P>) {
        const { context: propsContext, ...others } = props;
        return (
            <ContextLayout>
                {context => (
                    <ContextProvider {...{ ...context, ...propsContext }}>
                        <Component {...(others as P)} />
                    </ContextProvider>
                )}
            </ContextLayout>
        );
    };
}
