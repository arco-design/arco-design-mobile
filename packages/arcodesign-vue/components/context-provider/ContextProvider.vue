<script lang="ts" setup>
import { provide } from 'vue';
import { GlobalContextParams } from './type';
import { configProviderInjectionKey, defaultContext } from './utils';

const props = withDefaults(defineProps<GlobalContextParams>(), defaultContext);
provide(configProviderInjectionKey, props);
</script>
<template>
    <slot />
</template>
