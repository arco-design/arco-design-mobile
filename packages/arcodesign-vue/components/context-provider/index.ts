import ContextProvider from './ContextProvider.vue';
import { installFactory } from '../_helpers/install';

export type * from './type';
export * from './utils';

const install = installFactory(ContextProvider);

export { ContextProvider };

export default {
    install,
};
