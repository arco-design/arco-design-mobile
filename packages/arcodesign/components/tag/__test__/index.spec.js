import React from 'react';
import { mount } from 'enzyme';
import demoTest from '../../../tests/demoTest';
import mountTest from '../../../tests/mountTest';
import { defaultContext } from '../../context-provider';
import Tag from '..';

const prefix = `${defaultContext.prefixCls}-tag`;

demoTest('tag');

mountTest(Tag, 'Tag');

describe('Tag', () => {
    mountTest(() => <Tag type="hollow">Tag</Tag>, 'should mount correctly when set type prop');
    mountTest(() => <Tag size="small">Tag</Tag>, 'should mount correctly when set size prop');
    it('should callback correctly when close icon be clicked', () => {
        const mockFn = jest.fn();
        const component = mount(<Tag closeable onClose={mockFn}>标签</Tag>);
        expect(component.find('.tag-close-wrap').length).toBe(1);
        component.find('.tag-close-wrap').simulate('click');
        expect(mockFn).toBeCalled();
    });
    it('should callback correctly when click tag', () => {
        const mockFn = jest.fn();
        const component = mount(<Tag onClick={mockFn}>标签</Tag>);
        component.simulate('click');
        expect(mockFn).toBeCalled();
    });
    it('should render correctly when set filleted/halfBorder/borderStyle', () => {
        const component = mount(<Tag filleted halfBorder borderStyle="solid">标签</Tag>);
        expect(component.find(`.${prefix}`).hasClass('half-border')).toBe(true);
        expect(component.find(`.${prefix}`).props().style.borderStyle).toBe('solid');
        expect(component.find(`.${prefix}`).hasClass('filleted')).toBe(true);
    });
    it('should render correctly when use tag list', () => {
        const list = [1, 1].map(item => ({
            closeable: true,
            children: `标签${item}`
        }));
        const component = mount(<Tag.List list={list}/>);
        expect(component.find(`.${prefix}`).length).toBe(3);
        expect(component.find('.tag-list-add-wrap').length).toBe(1);
    });
})
