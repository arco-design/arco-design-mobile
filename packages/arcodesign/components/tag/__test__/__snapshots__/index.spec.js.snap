// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tag demo test tag demo: border.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="demo-tag-border"
  >
    <div
      class="arco-tag all-border-box arco-tag-hollow pc size-small"
      style="color: rgb(22, 93, 255); border-color: #94bfff; border-style: solid;"
    >
      <span
        class="tag-text"
      >
        Tag
      </span>
    </div>
    <div
      class="arco-tag all-border-box arco-tag-hollow half-border pc size-small"
      style="color: rgb(0, 180, 42); border-color: #7be188; border-style: solid;"
    >
      <span
        class="tag-text"
      >
        Tag
      </span>
    </div>
    <div
      class="arco-tag all-border-box arco-tag-hollow pc size-small"
      style="color: rgb(255, 125, 0); border-color: #ffcf8b; border-style: dashed;"
    >
      <span
        class="tag-text"
      >
        Tag
      </span>
    </div>
    <div
      class="arco-tag all-border-box arco-tag-hollow half-border pc size-small"
      style="color: rgb(245, 63, 63); border-color: #fbaca3; border-style: dashed;"
    >
      <span
        class="tag-text"
      >
        Tag
      </span>
    </div>
    <div
      class="arco-tag all-border-box arco-tag-hollow half-border pc size-small"
      style="color: rgb(78, 89, 105); border-color: #c9cdd4; border-style: dotted;"
    >
      <span
        class="tag-text"
      >
        Tag
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`tag demo test tag demo: custom-color.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-tag all-border-box arco-tag-solid half-border pc size-medium"
    style="background: rgb(255, 87, 34); border-style: solid;"
  >
    <span
      class="tag-text"
    >
      Tag
    </span>
  </div>
  <div
    class="arco-tag all-border-box arco-tag-solid half-border pc size-medium"
    style="border-style: solid;"
  >
    <span
      class="tag-text"
    >
      Tag
    </span>
  </div>
  <div
    class="arco-tag all-border-box arco-tag-primary half-border pc size-medium"
    style="color: rgb(15, 198, 194); background: rgb(232, 255, 251); border-style: none;"
  >
    <span
      class="tag-text"
    >
      Tag
    </span>
  </div>
  <div
    class="arco-tag all-border-box arco-tag-primary half-border pc size-medium"
    style="color: rgb(183, 29, 232); background: rgb(253, 232, 255); border-style: none;"
  >
    <span
      class="tag-text"
    >
      Tag
    </span>
  </div>
</DocumentFragment>
`;

exports[`tag demo test tag demo: custom-style.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-tag all-border-box arco-tag-solid half-border pc size-small"
    style="background: rgb(245, 63, 63); border-style: solid; border-radius: 0 4px 0 4px; height: 16px;"
  >
    <span
      class="tag-text"
    >
      Hot
    </span>
  </div>
</DocumentFragment>
`;

exports[`tag demo test tag demo: filleted.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-tag all-border-box arco-tag-solid half-border pc size-medium filleted"
    style="border-style: solid;"
  >
    <span
      class="tag-text"
    >
      Tag
    </span>
  </div>
  <div
    class="arco-tag all-border-box arco-tag-solid half-border pc size-medium filleted"
    style="background: rgb(0, 180, 42); border-style: solid;"
  >
    <span
      class="tag-text"
    >
      Tag
    </span>
  </div>
  <div
    class="arco-tag all-border-box arco-tag-solid half-border pc size-medium filleted"
    style="background: rgb(255, 125, 0); border-style: solid;"
  >
    <span
      class="tag-text"
    >
      Tag
    </span>
  </div>
  <div
    class="arco-tag all-border-box arco-tag-solid half-border pc size-medium filleted"
    style="background: rgb(245, 63, 63); border-style: solid;"
  >
    <span
      class="tag-text"
    >
      Tag
    </span>
  </div>
  <div
    class="arco-tag all-border-box arco-tag-solid half-border pc size-medium filleted"
    style="background: rgb(201, 205, 212); border-style: solid;"
  >
    <span
      class="tag-text"
    >
      Tag
    </span>
  </div>
</DocumentFragment>
`;

exports[`tag demo test tag demo: icon.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-tag all-border-box arco-tag-primary half-border pc size-medium"
    style="border-style: none;"
  >
    <svg
      class="arco-icon arco-icon-star "
      height="1em"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M268.437 941.29a35.563 35.563 0 01-50.773-35.733l25.899-248.49a35.563 35.563 0 00-8.896-27.456L66.176 442.155a35.563 35.563 0 0119.008-58.539l246.507-52.608a35.563 35.563 0 0023.36-16.96L481.237 95.872a35.563 35.563 0 0161.568 0l126.187 218.176a35.525 35.525 0 0023.36 16.96l246.507 52.608a35.563 35.563 0 0119.008 58.539L789.376 629.61a35.533 35.533 0 00-8.917 27.456l25.898 248.49a35.563 35.563 0 01-50.773 35.734l-228.16-109.739a35.563 35.563 0 00-30.805 0l-228.16 109.739zm191.147-186.645a120.896 120.896 0 01104.81 0l147.84 71.104-16.682-159.829a120.896 120.896 0 0130.336-93.355L836.864 449.11l-162.347-34.645a120.896 120.896 0 01-79.402-57.685L512 213.099l-83.115 143.68a120.896 120.896 0 01-79.402 57.685l-162.347 34.645 110.976 123.456a120.896 120.896 0 0130.336 93.355l-16.683 159.83 147.84-71.105z"
        fill="currentColor"
      />
    </svg>
    <span
      class="tag-text"
    >
      Tag
    </span>
  </div>
  <div
    class="arco-tag all-border-box arco-tag-primary half-border pc size-medium"
    style="color: rgb(255, 125, 0); background: rgb(255, 247, 232); border-style: none;"
  >
    <svg
      class="arco-icon arco-icon-notice "
      fill="currentColor"
      height="1em"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <defs />
      <path
        d="M618.667 853.333A21.333 21.333 0 01640 874.667v42.666a21.333 21.333 0 01-21.333 21.334H405.333A21.333 21.333 0 01384 917.333v-42.666a21.333 21.333 0 0121.333-21.334h213.334zM576 64a21.333 21.333 0 0121.333 21.333v32.427c147.222 39.125 256 177.067 256 341.248V704h64a21.333 21.333 0 0121.334 21.333V768a21.333 21.333 0 01-21.334 21.333H106.667A21.333 21.333 0 0185.333 768v-42.667A21.333 21.333 0 01106.667 704h64V459.008c0-164.181 108.8-302.123 256-341.248V85.333A21.333 21.333 0 01448 64h128zm-64 128c-141.376 0-256 119.595-256 267.136V704h512V459.136C768 311.595 653.376 192 512 192z"
      />
    </svg>
    <span
      class="tag-text"
    >
      Tag
    </span>
  </div>
</DocumentFragment>
`;

exports[`tag demo test tag demo: index.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-tag all-border-box arco-tag-primary half-border pc size-small"
    style="border-style: none;"
  >
    <span
      class="tag-text"
    >
      Tag
    </span>
  </div>
  <div
    class="arco-tag all-border-box arco-tag-primary half-border pc size-small"
    style="color: rgb(0, 180, 42); background: rgb(232, 255, 234); border-style: none;"
  >
    <span
      class="tag-text"
    >
      Tag
    </span>
  </div>
  <div
    class="arco-tag all-border-box arco-tag-primary half-border pc size-small"
    style="color: rgb(255, 125, 0); background: rgb(255, 247, 232); border-style: none;"
  >
    <span
      class="tag-text"
    >
      Tag
    </span>
  </div>
  <div
    class="arco-tag all-border-box arco-tag-primary half-border pc size-small"
    style="color: rgb(245, 63, 63); background: rgb(255, 236, 232); border-style: none;"
  >
    <span
      class="tag-text"
    >
      Tag
    </span>
  </div>
  <div
    class="arco-tag all-border-box arco-tag-primary half-border pc size-small"
    style="color: rgb(78, 89, 105); background: rgb(242, 243, 245); border-style: none;"
  >
    <span
      class="tag-text"
    >
      Tag
    </span>
  </div>
  <div
    class="arco-tag all-border-box arco-tag-primary half-border pc size-small"
    style="color: rgb(134, 144, 156); background: rgb(242, 243, 245); border-style: none;"
  >
    <span
      class="tag-text"
    >
      Disabled
    </span>
  </div>
</DocumentFragment>
`;

exports[`tag demo test tag demo: list.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-tag-list"
    style="margin-bottom: -10px;"
  >
    <div
      class="arco-tag all-border-box arco-tag-primary half-border pc size-medium"
      style="color: rgb(22, 93, 255); background: rgb(232, 243, 255); border-style: none; margin-bottom: 10px;"
    >
      <span
        class="tag-text"
      >
        Tag1
      </span>
      <div
        class="tag-close-wrap"
      >
        <svg
          class="arco-icon arco-icon-close tag-close-icon"
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M509.568 447.083L796.16 160.469a21.333 21.333 0 0130.165 0l30.166 30.166a21.333 21.333 0 010 30.186L569.899 507.435l286.613 286.592a21.333 21.333 0 010 30.165l-30.165 30.165a21.333 21.333 0 01-30.166 0L509.568 567.765 222.933 854.38a21.333 21.333 0 01-30.186 0l-30.166-30.166a21.333 21.333 0 010-30.165l286.614-286.613L162.603 220.8a21.333 21.333 0 010-30.187l30.165-30.165a21.333 21.333 0 0130.187 0l286.613 286.613z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      class="arco-tag all-border-box arco-tag-primary half-border pc size-medium"
      style="color: rgb(0, 180, 42); background: rgb(232, 255, 234); border-style: none; margin-bottom: 10px;"
    >
      <span
        class="tag-text"
      >
        Tag2
      </span>
      <div
        class="tag-close-wrap"
      >
        <svg
          class="arco-icon arco-icon-close tag-close-icon"
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M509.568 447.083L796.16 160.469a21.333 21.333 0 0130.165 0l30.166 30.166a21.333 21.333 0 010 30.186L569.899 507.435l286.613 286.592a21.333 21.333 0 010 30.165l-30.165 30.165a21.333 21.333 0 01-30.166 0L509.568 567.765 222.933 854.38a21.333 21.333 0 01-30.186 0l-30.166-30.166a21.333 21.333 0 010-30.165l286.614-286.613L162.603 220.8a21.333 21.333 0 010-30.187l30.165-30.165a21.333 21.333 0 0130.187 0l286.613 286.613z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
    <div
      class="tag-list-add-wrap"
    >
      <div
        class="arco-tag all-border-box tag-list-add arco-tag-primary half-border pc size-medium"
        style="border-style: dashed; margin-bottom: 10px;"
      >
        <svg
          class="arco-icon arco-icon-add "
          height="1em"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M527.317 85.333c9.515 0 12.95.982 16.427 2.859a19.337 19.337 0 018.064 8.064c1.877 3.477 2.859 6.912 2.859 16.427v356.65h356.65c9.515 0 12.95.982 16.427 2.859a19.337 19.337 0 018.064 8.064c1.877 3.477 2.859 6.912 2.859 16.427v30.634c0 9.515-.982 12.95-2.859 16.427a19.392 19.392 0 01-8.064 8.064c-3.477 1.877-6.912 2.859-16.427 2.859h-356.65v356.65c0 9.515-.982 12.95-2.859 16.427a19.392 19.392 0 01-8.064 8.064c-3.477 1.877-6.912 2.859-16.427 2.859h-30.634c-9.515 0-12.95-.982-16.427-2.859a19.392 19.392 0 01-8.064-8.064c-1.877-3.477-2.859-6.912-2.859-16.427l-.021-356.65h-356.63c-9.514 0-12.949-.982-16.426-2.859a19.392 19.392 0 01-8.064-8.064c-1.877-3.477-2.859-6.912-2.859-16.427v-30.634c0-9.515.982-12.95 2.859-16.427a19.392 19.392 0 018.064-8.064c3.477-1.877 6.912-2.859 16.427-2.859l356.65-.021v-356.63c0-9.514.982-12.949 2.859-16.426a19.392 19.392 0 018.064-8.064c3.477-1.877 6.912-2.859 16.427-2.859h30.634z"
            fill="currentColor"
          />
        </svg>
        <span
          class="tag-text"
        >
          添加标签
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`tag demo test tag demo: size.md renders correctly 1`] = `
<DocumentFragment>
  <div
    class="arco-tag all-border-box arco-tag-primary half-border pc size-small"
    style="border-style: none;"
  >
    <span
      class="tag-text"
    >
      Tag
    </span>
  </div>
  <div
    class="arco-tag all-border-box arco-tag-primary half-border pc size-medium"
    style="border-style: none;"
  >
    <span
      class="tag-text"
    >
      Tag
    </span>
  </div>
  <div
    class="arco-tag all-border-box arco-tag-primary half-border pc size-large"
    style="border-style: none;"
  >
    <span
      class="tag-text"
    >
      Tag
    </span>
  </div>
</DocumentFragment>
`;
